### ISSUE TITLE
Updating existing mod folders with steamcmd does not delete outdated files

### ISSUE THREAD

**Issue by Xetaxheb** [rimpy-custom/RimPy#143]
As here: https://github.com/lilwhitemouse/RimWorld-LWM.DeepStorage/issues/144#issuecomment-1704355643

TL;DR if you use the downloader to update a mod or the processes that in turn use the downloader and therefore steamcmd, outdated files (like a removed LoadOrder.xml) do not get purged.

Whether this is a steamcmd issue or a rimpy issue I can almost guarantee you things won't be changed on steamcmd's end because breaking compatibility, 
so it might be worth to purge a modfolder and redownload fully in any instance where the folder exists ?

Short of extended tracking/comparing to outdated workshop downloads, this is probably the best solution at the risk of deleting customized files inside of it (say, tweaked or created xml files) by the end user. (I'm not aware of any mods that save data in the modfolder themselves? could be ig? though they shouldn't be)

Could maybe make a yes/no for whether to purge existing files (recommended yes).

**Comment by dougbenham** [comment]
I created this https://github.com/dougbenham/SteamCmdWrapper maybe it will be of use to you.
